main.go

```diff
@@ -52,44 +52,16 @@ func main() {
 			logger.Fatalf("Error querying snap store: %s", err)
 		}
 
-		// launchpad
-		builds, err := queryLaunchpad(k)
-		if err != nil {
-			logger.Fatalf("Error querying launchpad: %s", err)
-		}
-		revisionBuildStatus := make(map[uint]string)
-		for _, v := range builds.Entries {
-			// Setting a check mark only if we find the successful build result for a given revision.
-			// Alternative scenarios include results that have no revision number because:
-			// - build or artifact upload has failed (an actual failure)
-			// - build is too old and not returned in the query
-			// - build or artifact upload is pending
-			if v.StoreUploadRevision != nil && v.BuildState == "Successfully built" {
-				revisionBuildStatus[*v.StoreUploadRevision] = "✅"
-			}
-		}
-
 		// github
 		runs, err := queryGithub(v.GithubRepo)
 		if err != nil {
 			logger.Fatalf("Error querying launchpad: %s", err)
 		}
-		var totalSnapRuns, failedSnapRuns uint
-		testIcon := "🔴"
-		for _, run := range runs.WorkflowRuns {
-			if run.Name == "Snap Testing" {
-				totalSnapRuns++
-			}
-			if run.Conclusion == "failure" {
-				failedSnapRuns++
-				logger.Errorf("🔴 %s (%s)", run.DisplayTitle, run.HTMLURL)
-			}
-		}
-		if totalSnapRuns == 0 { // something is not right
-			testIcon = "🟠"
-		} else if failedSnapRuns == 0 {
-			testIcon = "🟢"
-		}
+		var build, test snapStatistcs
+		build.runName = "Snap Builder"
+		test.runName = "Snap Testing"
+		handleRun(runs, &build)
+		handleRun(runs, &test)
```

Comment:
This looks very much like C coding style, which Go allows, but it is discouraged as there are better ways to do this with superior readability and maintainability.
I suggest making those functions accept the filter argument (run name) and return snapStatistics. The function name should match its functionality, ideally based on what it takes and returns, e.g.:
```go
func computeWorkflowRunStats(runs *runs, runName string) workflowRunStats
```

Reply:
Also done!